client/app/src/main/java/com/myproj/client/fragments/ChartFragment.kt

```diff
@@ -109,44 +104,24 @@ class ChartFragment : Fragment() {
         lineChart.invalidate()
     }
 
-
-    private fun getSensorData(command: String, param1: String, sampleList: MutableList<CO2Sample>) {
-        val comm = Command(command, param1)
-        apiService.getSensorData(comm).enqueue(object : Callback<ResponseBody> {
-            @SuppressLint("NotifyDataSetChanged")
-            override fun onResponse(call: Call<ResponseBody>, response: Response<ResponseBody>) {
-                if (response.isSuccessful) {
-                    response.body()?.let { body ->
-                        val json = body.string()
-                        if (json.isNotEmpty()) {
-                            val jsonArray = JSONArray(json)
-                            sampleList.clear()
-                            for (i in 0 until jsonArray.length()) {
-                                val jsonObject = jsonArray.getJSONObject(i)
-                                val datetime = jsonObject.getString("datetime")
-                                val co2Level = jsonObject.getString("CO2Level")
-                                val co2Sample = CO2Sample(datetime, co2Level)
-                                sampleList.add(co2Sample)
-                            }
-                            requestsCompleted++
-                            if (requestsCompleted == 2) {
-                                drawChart(samplesIndoor, samplesOutdoor)
-                            }
-                        } else {
-                            // Обробка порожньої відповіді JSON
-                        }
-                    }
-                } else {
-                    // Обробка неуспішної відповіді
+    private fun getSensorData(command: String, sampleList: MutableList<CO2Sample>) {
+        ApiClient.getSensorData(command, "", object : SensorDataCallback {
+            override fun onSuccess(samples: List<CO2Sample>) {
+                sampleList.clear()
+                sampleList.addAll(samples)
+                requestsCompleted++
+                if (requestsCompleted == 2) {
+                    drawChart(samplesIndoor, samplesOutdoor)
                 }
             }
 
-            override fun onFailure(call: Call<ResponseBody>, t: Throwable) {
-                // Обробка помилки
+            override fun onFailure(t: Throwable) {
+                // Handle failure
```

Comment:
At least log it or make a popup for user to see (message box with error)

Reply:
Done